### PR TITLE
feat: add reachable vulnerabilities support for gradle

### PR DESCRIPTION
**Pull request by miiila**
- [x] Ready for review
- [ ] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Adds reachable vulnerabilities feature support for Gradle projects.

⚠️ Will fail unless https://github.com/snyk/snyk-gradle-plugin/pull/147 is merged.

### CODE REVIEW

File: test/reachable-vulns.test.ts [hunk 1]
```diff
@@ -166,9 +167,9 @@ test('formatReachablePaths', (t) => {
 });
 
 test('validatePayload - not supported package manager', async (t) => {
-  const pkgManagers = Object.keys(SUPPORTED_PACKAGE_MANAGER_NAME);
-  const mavenIndex = pkgManagers.indexOf('maven');
-  pkgManagers.splice(mavenIndex, 1); // remove maven as it's supported
+  const pkgManagers = Object.keys(SUPPORTED_PACKAGE_MANAGER_NAME).filter(
+    (name: any) => !REACHABLE_VULNS_SUPPORTED_PACKAGE_MANAGERS.includes(name),
```
Comment (anthogez) [inline test/reachable-vulns.test.ts:171]: I suggest `name: string` instead of `name: any`

File: package.json [hunk 2]
```diff
@@ -79,7 +79,7 @@
     "snyk-cpp-plugin": "2.0.0",
     "snyk-docker-plugin": "3.24.0",
     "snyk-go-plugin": "1.16.2",
-    "snyk-gradle-plugin": "3.6.3",
+    "snyk-gradle-plugin": "3.7.0",
```
Comment (anthogez) [inline package.json:82]: reviewed also changes introduced here
https://github.com/snyk/snyk-gradle-plugin/commit/453a81ec1d68297a426179163aea09d29022892a

**Review by anthogez — APPROVED** [review]
My only concern is the type the rest is good to me

File: test/reachable-vulns.test.ts (hunk 1 above)
Comment (darmalovan) [reply test/reachable-vulns.test.ts]: Fixed to the right type!

Comment (anthogez) [reply test/reachable-vulns.test.ts]: Even better, perfect 👌 

image

**Review by JackuB — APPROVED** [review]

**Pull request merged**